patroni/dcs/etcd.py

```diff
@@ -451,25 +466,32 @@ def _prepare_request(self, kwargs, params=None, method=None):
 
 class AbstractEtcd(AbstractDCS):
 
-    def __init__(self, config, client_cls, retry_errors_cls):
+    def __init__(self, config: Dict[str, Any], client_cls: Type[AbstractEtcdClientWithFailover],
+                 retry_errors_cls: Union[Type[Exception], Tuple[Type[Exception], ...]]) -> None:
         super(AbstractEtcd, self).__init__(config)
         self._retry = Retry(deadline=config['retry_timeout'], max_delay=1, max_tries=-1,
                             retry_exceptions=retry_errors_cls)
         self._ttl = int(config.get('ttl') or 30)
-        self._client = self.get_etcd_client(config, client_cls)
+        self._abstract_client = self.get_etcd_client(config, client_cls)
         self.__do_not_watch = False
         self._has_failed = False
 
-    def reload_config(self, config):
+    @property
+    @abc.abstractmethod
+    def _client(self) -> AbstractEtcdClientWithFailover:
+        """return correct type of etcd client"""
+
+    def reload_config(self, config: Union['Config', Dict[str, Any]]) -> None:
         super(AbstractEtcd, self).reload_config(config)
         self._client.reload_config(config.get(self.__class__.__name__.lower(), {}))
 
-    def retry(self, *args, **kwargs):
+    def retry(self, *args: Any, **kwargs: Any) -> Any:
```

Comment:
Should this method have a `func` arg as the first arg to match the Retry class `__call__`?